### PR TITLE
[10.x] add Storage::json() method to read and decode a json file

### PR DESCRIPTION
**Pull request by lorenzolosa**
Laravel v10.4.0 introduced a `File::json()` method to retrieve json encoded data from a file. That's nice, but often files are accessed using `Storage` and not `File`. Here I'm adding a similar `Storage::json()`, i.e.:

```
$data = Storage::json('sample.json');
```

which is equivalent to:

```
$contents = Storage::get('sample.json');
$data = json_decode($contents, true);
```

This is really the same as #46481, but for `Storage` instead of `File`.

### CODE REVIEW

**Comment by ankurk91** [comment]
Can you add JSON_THROW_ON_ERROR option

**Comment by taylorotwell** [comment]
Flag support added to both methods.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -63,14 +63,15 @@ public function get($path, $lock = false)
      * Get the contents of a file as decoded JSON.
      *
      * @param  string  $path
+     * @param  int  $flags
      * @param  bool  $lock
      * @return array
      *
      * @throws \Illuminate\Contracts\Filesystem\FileNotFoundException
      */
-    public function json($path, $lock = false)
```
Comment (chrysanthos) [inline src/Illuminate/Filesystem/Filesystem.php:71]: Isn't this a breaking change? The $lock parameter is now third instead of second?

**Comment by driesvints** [comment]
I've sent in a PR to revert this one.

**Pull request merged**